src/main/scala/User.scala

```diff
@@ -1,3 +1,120 @@
-object User {
+import DataBaseConn.{getAll, putUser}
+import User.registerAndLoginRoute
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{ContentTypes, HttpEntity, StatusCodes}
+import akka.http.scaladsl.server.Directives._
+import spray.json._
 
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+/*
+  POST /register -> register User(username, password)
+  GET /users -> get all the users
+  GET /user/{id} -> get a user by id
+  GET /login -> OK (Successful), NotFound (Error)
+ */
+
+// id is auto incremented
+case class UserComplete(id: Int, username: String, password: String)
+case class UserPartial(username: String, password: String)
+
+trait UserJsonProtocol extends DefaultJsonProtocol {
+  implicit val userPartialFormat = jsonFormat2(UserPartial)
+  implicit val userCompleteFormat = jsonFormat3(UserComplete)
+}
+
+trait ActorUser{
+  implicit val system = ActorSystem("UserActor")
+}
+
+object User extends UserJsonProtocol with ActorUser {
+
+  import system.dispatcher
+
+  def checkLoginUser(username: String, password: String): Boolean = {
+    if(!(getAll.find(i => i.username == username && i.password == password).isEmpty)) true
```

Comment:
Can we get rid of this since we have DB connection now?

Reply:
I used this to check if the user who wants to authenticate has an account, it is in the database. Was it better if i just looked it up in the database by his name and password, right?

Reply:
Yes, you can just check the credentials into the DB (i.e. check for a combination of username + pass if it returns exactly 1 row, here you go, you have a genuine login)